packages/loot-core/src/server/budget/goals/goalsSchedule.ts

```diff
@@ -90,6 +89,10 @@ export async function goalsSchedule(
                 dateConditions,
                 monthUtils._parse(next_date),
               );
+              if (!diffDays) {
```

Comment:
> I don't think this works correctly. If you have a schedule that has two occurrences in the same month, the function could exit prematurely. I think diffDays needs to be reevaluated before doing this check.

Oh, nice catch! This should definitely have been reevaluated. I decided to slightly update how the logic works to be more straightforward/correct in general, let me know your thoughts! Also fixed another bug I found incidentally, that one can move to a separate PR if you like.

Reply:
My test budget is crashing with the current build.  Here it is if you want to try and debug it.  I believe it was working before these most recent changes, so its likely somewhere in there.
[template_test_export.zip](https://github.com/actualbudget/actual/files/13547319/template_test_export.zip)

Reply:
Ah yikes, sorry about that. I'll look into this later today.l and add what I find to my own manual tests.

Reply:
Alright, sorry for the pain here. I was able to run your test budget with no issues in the latest commit.

Reply:
I've attached a test budget.
[2023-12-05-My-Finances-564f13f.zip](https://github.com/actualbudget/actual/files/13573272/2023-12-05-My-Finances-564f13f.zip)

In this test file if you use the "Savings" category that has a schedule called "breakWeekend", the months of December and March aren't right.  Those days both fall on a Saturday those months and only budget for one of the two schedules.  
I haven't tried it, but I think this only occurs on the 'before' style schedules.  What if instead of passing the weekendSolveMode variable of the actual schedule, we force it to 'after' so diffDays won't automatically be evaluated as 0 and will then look for a second schedule for the month?  Thanks for all the cleanup you've done!

Reply:
Oh, nice catch! This was just one place I missed in my refactor, I believe this should be fixed in that budget now.

Reply:
I think you've got it!